### PR TITLE
Always allow and support quoted subscript expressions

### PR DESCRIPTION
**Pull request by seut**
Add full support for quoted subscript expressions by default.
Partial quoted subscript expressions will work as well.
    
This adds out-of-the-box support for various clients which do not know CrateDB’s special subscript expression (and also don’t make use of the server-side `quote_ident` function) and thus always quote the full
qualified column name.
Inner columns (schema) of object data types are exposed by system informational tables at ``information_schema``, ``sys`` and ``pg_catalog`` schemas and thus used by clients.

Relates #10163.

### CODE REVIEW

**Comment by seut** [comment]
@mfussenegger Could you please have rough look on this? I think this could be even ready for review if we agree on this feature. The complexity for this is quite low, it will not introduce any BWC or performance issues (as far as I know), so I'd tend to add it.

As discussed the other option to add support for problematic clients like PowerBI would be #10485.
Although this PR would not solve the root cause of #10148 which we would still have to solve by e.g. giving each sub-column a dedicated `ordinal_position`.

Another reason why I prefer quoted subscript support over disabling the exposure of sub-columns is that our object sub-columns are somehow first-class citizens as we support to use them at `primary key`, `partition by`, `clustered by` etc clauses. Disable exposure of sub-columns in these context is no option afaik as it would hide important schema information. 
And disabling exposure of them partially in some contexts (like `information_schema.columns`) seems weird and somehow broken.

**Comment by mfussenegger** [comment]
@seut yeah I tend to agree. Complexity seems okay and I think for users this will result in the better experience. So I'd say let's go ahead with this solution.


> which we would still have to solve by e.g. giving each sub-column a dedicated ordinal_position.

Maybe we start treating them like normal columns and increment the counter for them? Though we have to be careful how that may affect BWC.

**Comment by seut** [comment]
@mfussenegger 

> Maybe we start treating them like normal columns and increment the counter for them? Though we have to be careful how that may affect BWC.

Yes this is what I'd suggest. I tried this already once and did some rough minimal tests (running test suite, connected with PowerBI, Grafana, etc) and did not run into issues. But of course a lot of BWC unknowns may exists.

File: server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java [hunk 1]
```diff
@@ -1202,6 +1175,16 @@ private static void ensureResultTypesMatch(Collection<? extends Symbol> results)
         }
     }
 
+    @Nullable
+    private static String maybeQuotedSubscript(String columnName) {
```
Comment (kovrus) [inline server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java:1179]:
```suggestion
    private static String maybeQuoteSubscript(String columnName) {
```

File: server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java [hunk 2]
```diff
@@ -676,6 +642,15 @@ protected Symbol visitSubscriptExpression(SubscriptExpression node, ExpressionAn
                 Symbol index = node.index().accept(this, context);
                 return allocateFunction(SubscriptFunction.NAME, List.of(base, index), context);
             } else {
+                // Detect and process partial quoted subscript expression
+                var nameParts = qualifiedName.getParts();
+                var columnName = nameParts.get(nameParts.size() - 1);
```
Comment (kovrus) [inline server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java:647]: maybe?
```suggestion
                var columnName = qualifiedName.getSuffix();
```

**Review by kovrus — APPROVED** [review]
otherwise lgtm

File: server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java (hunk 1 above)
Comment (seut) [reply server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java:1179]: The function will detect a quoted subscript and re-add quoting around the column identifier. So it is not maybe quoting.
The existing name is also confusing and not really correct, new suggestion: `detectAndSanitizeQuotedSubscript`
WDYT?

Comment (kovrus) [reply server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java:1179]: `detectAndSanitizeQuotedSubscript` sounds better, `maybeQuotedSubscript` was a bit confusing

**Pull request merged**